### PR TITLE
use GenericOperation to simplify vm operation

### PR DESCRIPTION
**Pull request by gao-feng**

### CODE REVIEW

**Comment by laijs** [comment]
please don't close the channel!

https://groups.google.com/forum/#!msg/golang-nuts/pZwdYRGxCIk/qpbHxRRPJdUJ
```
Note that it is only necessary to close a channel if the receiver is
looking for a close.  Closing the channel is a control signal on the
channel indicating that no more data follows.
```

File: hypervisor/vm.go [hunk 1]
```diff
@@ -421,57 +421,64 @@ func (vm *Vm) KillContainer(container string, signal syscall.Signal) error {
...
-				return
-			}
-			result <- fmt.Errorf("allocate device failed")
-			return
-		default:
-			result <- fmt.Errorf("get unexpected event %s", EventString(ev.Event()))
-			return
-		}
+	client := make(chan VmEvent, 1)
+	defer close(client)
```
Comment (laijs) [inline hypervisor/vm.go]: don't close the channel in receiver side.

**Comment by laijs** [comment]
an example
```go
package promise

// Go is a basic promise implementation: it wraps calls a function in a goroutine,
// and returns a channel which will later return the function's return value.
func Go(f func() error) chan error {
	ch := make(chan error, 1)
	go func() {
		ch <- f()
	}()
	return ch
}
```

File: hypervisor/vm.go (hunk 1 above)
Comment (gao-feng) [inline hypervisor/vm.go]: get, will update

**Comment by gao-feng** [comment]
do we really need promise for SendGenericOperation?

**Comment by laijs** [comment]
promise is an example that the channel is not closed after used.

**Comment by gao-feng** [comment]
ok, get it

**Pull request merged**